meegkit/utils/trca.py

```diff
@@ -0,0 +1,135 @@
+
+    The rules are:
+    from 0 to 4 rounds down
+    from 5 to 9 rounds up
+
+    Parameters
+    ----------
+    num : float
+        Number to round
+    decimals : number of decimals
+
+    Returns
+    -------
+    num rounded
+    """
+    multiplier = 10 ** decimals
+    return int(np.floor(num * multiplier + 0.5) / multiplier)
+
+
+def normfit(data, ci=0.95):
+    """Compute the mean, std and confidence interval for them.
+
+    Parameters
+    ----------
+    data: array, shape=()
+        Input data.
+    ci : float
+        Confidence interval (default=0.95).
+
+    Returns
+    -------
+    m : mean
+    sigma : std deviation
+    [m - h, m + h] : confidence interval of the mean
+    [sigmaCI_lower, sigmaCI_upper] : confidence interval of the std
+    """
+    arr = 1.0 * np.array(data)
+    num = len(arr)
+    avg, std_err = np.mean(arr), scipy.stats.sem(arr)
+    h_int = std_err * t.ppf((1 + ci) / 2., num - 1)
+    var = np.var(data, ddof=1)
+    var_ci_upper = var * (num - 1) / (chi2.ppf((1 - ci) / 2, num - 1))
+    var_ci_lower = var * (num - 1) / (chi2.ppf(1 - (1 - ci) / 2, num - 1))
+    sigma = np.sqrt(var)
+    sigma_ci_lower = np.sqrt(var_ci_lower)
+    sigma_ci_upper = np.sqrt(var_ci_upper)
+
+    return avg, sigma, [avg - h_int, avg +
+                        h_int], [sigma_ci_lower, sigma_ci_upper]
+
+
+def itr(n, p, t):
+    """Compute information transfer rate (ITR).
+
+    Inputs
+    ------
+    n : int
+        Number of targets.
+    p : float
+        Target identification accuracy (0 <= p <= 1).
+    t : float
+        Average time for a selection (s).
+
+    Returns
+    -------
+    itr : float
+        Information transfer rate [bits/min]
+
+    References
+    ----------
+    .. [1] M. Cheng, X. Gao, S. Gao, and D. Xu,
+        "Design and Implementation of a Brain-Computer Interface With High
+        Transfer Rates", IEEE Trans. Biomed. Eng. 49, 1181-1186, 2002.
+
+    """
+    itr = 0
+
+    if (p < 0 or 1 < p):
+        raise ValueError('Accuracy need to be between 0 and 1.')
+    elif (p < 1 / n):
+        raise ValueError('ITR might be incorrect because accuracy < chance')
+        itr = 0
+    elif (p == 1):
+        itr = np.log2(n) * 60 / t
+    else:
+        itr = (np.log2(n) + p * np.log2(p) + (1 - p) *
+               np.log2((1 - p) / (n - 1))) * 60 / t
+
+    return itr
+
+
+def filterbank(eeg, fs, idx_fb):
+    """Filter bank design for decomposing EEG data into sub-band components.
+
+    Parameters
+    ----------
+    eeg : np.array, shape=(n_samples, n_chans[, n_trials])
+        Training data.
+    fs : int
+        Sampling frequency of the data.
+    idx_fb : int
+        Index of filters in filter bank analysis
+
+    Returns
+    -------
+    y: np.array, shape=(n_trials, n_chans, n_samples)
+        Sub-band components decomposed by a filter bank.
+
+    """
+    fs = fs / 2
+    passband = [6, 14, 22, 30, 38, 46, 54, 62, 70, 78]
```

Comment:
For a more general-purpose/use of the filterbank function, Passband and Stopband should be passed -as parameters in the function. 
In fact, the indicated values are related to Nakanishi exp/tutorial with an interval of frequencies of the visual stimuli between 8-15Hz.

So everyone who wants to use the filterbank function should change those values with new values that will allow a better capturing of the activity of the harmonics related to the frequencies of the stimuli of his own experiment.

Please, let me know if is clear what I mean.

Reply:
Ah yes you're right, I need to expose a filterbank argument in the TRCA class.